### PR TITLE
TASK: Check against LiveWorkspaceName Instead Of Comparing CS Ids

### PR DESCRIPTION
**Pull request by mhsdesign**
With https://github.com/neos/neos-development-collection/pull/5002 `\Neos\Neos\FrontendRouting\Projection\DocumentUriPathFinder::isLiveContentStream` is obsolete.

See https://github.com/neos/neos-development-collection/issues/4996
> Afterwards, affected Neos projections (e.g. `ChangeProjection` and `UriPathProjection`) should be simplified such that they no longer keep track of the workspace

Adjustments:
- https://github.com/neos/redirecthandler-neosadapter/pull/78

### CODE REVIEW

File: Neos.Neos/Classes/FrontendRouting/Projection/DocumentUriPathSchemaBuilder.php [hunk 1]
```diff
@@ -22,8 +22,7 @@ public function __construct(
     public function buildSchema(AbstractSchemaManager $schemaManager): Schema
     {
         $schema = DbalSchemaFactory::createSchemaWithTables($schemaManager, [
-            $this->createUriTable(),
-            $this->createLiveContentStreamsTable()
```
Comment (nezaniel) [inline Neos.Neos/Classes/FrontendRouting/Projection/DocumentUriPathSchemaBuilder.php:37]: Also a more general question:
Wouldn't this be also responsible for now removing the deprecated table?

**Review by nezaniel — DISMISSED** [review]
one little thing, rest looks nice

**Comment by mhsdesign** [comment]
Updated branch and added migration to drop the tables

File: Neos.Neos/Classes/FrontendRouting/Projection/DocumentUriPathSchemaBuilder.php (hunk 1 above)
Comment (dlubitz) [reply Neos.Neos/Classes/FrontendRouting/Projection/DocumentUriPathSchemaBuilder.php:37]: Good question in general. Have the same issue with another table. But does it make sence to check for removal on each setup? How long? When will we remove the removal code?

**Review by dlubitz — APPROVED** [review]
Looks good to me by reading.

File: Neos.Neos/Classes/PendingChangesProjection/ChangeProjection.php [hunk 2]
```diff
@@ -490,12 +488,4 @@ private function getChange(
         // We always allow root nodes
         return $changeRow ? Change::fromDatabaseRow($changeRow) : null;
     }
-
-    private function isLiveContentStream(ContentStreamId $contentStreamId): bool
```
Comment (kitsunet) [inline Neos.Neos/Classes/PendingChangesProjection/ChangeProjection.php:494]: We never need to know this anymore? Weird if so, I would think this is quite useful to have.

**Review by kitsunet — APPROVED** [review]
Added comment, deleted comment, seems fine... Thanks!

File: Neos.Neos/Classes/FrontendRouting/Projection/DocumentUriPathProjection.php [hunk 3]
```diff
@@ -108,7 +107,12 @@ private function determineRequiredSqlStatements(): array
     {
         $schemaManager = $this->dbal->createSchemaManager();
         $schema = (new DocumentUriPathSchemaBuilder($this->tableNamePrefix))->buildSchema($schemaManager);
-        return DbalSchemaDiff::determineRequiredSqlStatements($this->dbal, $schema);
+        $statements = DbalSchemaDiff::determineRequiredSqlStatements($this->dbal, $schema);
+        // MIGRATIONS
+        if ($this->dbal->getSchemaManager()->tablesExist([$this->tableNamePrefix . '_livecontentstreams'])) {
+            $statements[] = sprintf('DROP table %s_livecontentstreams;', $this->tableNamePrefix);
```
Comment (kitsunet) [inline Neos.Neos/Classes/FrontendRouting/Projection/DocumentUriPathProjection.php:113]: not a huge fan.... I would just leave this dangling, we can consider something for these cases after release...

File: Neos.Neos/Classes/FrontendRouting/Projection/DocumentUriPathSchemaBuilder.php (hunk 1 above)
Comment (bwaidelich) [reply Neos.Neos/Classes/FrontendRouting/Projection/DocumentUriPathSchemaBuilder.php:37]: "Create" is a bit misleading here – it creates the schema for the table. There is no way to _remove_ a schema here.
That's currently done in the projection itself (see https://github.com/neos/neos-development-collection/pull/5149/files#diff-2e3df51a3340b51e08cf9d2ba5aade329a8c0e073354c1d5a72e42c3aaf5a127R112)
But I'm with @kitsunet on this one: Let's leave those dangling for now. For 9.0 -> 9.1 ect we'll have to think of a way to migrate those things once

**Review by bwaidelich — APPROVED** [review]
+1 by reading, thanks!
Re dropping the table I'm OK either way.

@nezaniel could you remove your -1 if you agree?

File: Neos.Neos/Classes/FrontendRouting/Projection/DocumentUriPathProjection.php (hunk 3 above)
Comment (mhsdesign) [reply Neos.Neos/Classes/FrontendRouting/Projection/DocumentUriPathProjection.php:113]: im not a huge fan either from these hardcoded migrations, but Sebastian started introducing them and we can delete them at some point if we want. So thats why ill merge it.

**Comment by mhsdesign** [comment]
Stupidly this introduced a regression: https://github.com/neos/neos-development-collection/pull/5293
As `$event->workspaceName` is not to be trusted.

**Pull request merged**